way_to_home/custom_user/views.py

```diff
@@ -63,3 +65,35 @@ def log_in(request):
         return HttpResponse('invalid credentials', status=400)
     login(request, user=user)
     return HttpResponse('operation was successful provided', status=200)
+
+
+@require_http_methods(["GET"])
+def auth_google(request):
+    """Function that provides getting url for confirmation access to user's data."""
+    google = OAuth2Session(client_id=CLIENT_ID, redirect_uri=REDIRECT_URI, state=STATE, scope=SCOPE)
+    data = google.authorization_url(url=AUTH_URL, state=STATE)[0]
+    if data:
+        return redirect(data)
+    return HttpResponse("Access denied", status=403)
+
+
+@require_http_methods(["GET"])
+def signin_google(request):
+    """Function that provides user registration or authorization via google."""
+    google = OAuth2Session(client_id=CLIENT_ID, redirect_uri=REDIRECT_URI, state=STATE, scope=SCOPE)
+    try:
+        code = request.GET["code"]
+    except ValueError:
+        return HttpResponse("Havent code", status=400)
```

Comment:
ValueError is odd in this case, consider checking if code == None

Reply:
also "Havent code" is confusing
consider changing variable name of **code** too

Reply:
done